### ISSUE TITLE
Keep output hidden

### ISSUE THREAD

**Issue by W4RH4WK** [saviorisdead/RustyCode#175]
This is more of a question. Is there a way to keep the output panel hidden when checking the file for errors. I mind popping it open when building, but when it just runs `cargo check` automatically on saving I'd rather have it keep the output panel hidden.

Error / Warning count in the lower left + underline squiggles are all I need.

Maybe this is already available and I just did not find it, please excuse me if this is the case.

**Comment by trixnz** [comment]
I strongly agree with this, the output window is useful if you're compiling in the IDE, but when you're saving files it gets a little tedious to keep closing the window. I personally disable the whole output window in a local branch as I compile outside of the IDE and have no need for it.

**Comment by Draivin** [comment]
I generally agree with this, but there is one related issue that I would like opinions about.

As an example, when running the check command, if there is more than one possible `bin`, cargo will fail and say that we should specify the appropriate `bin` . In such cases the error check will not produce any results, as the compilation won't even start.

Obviously we should add an option to specify the `bin` that should be used for cargo invocations, but I believe it would also be good to show some message to the user in case the cargo check didn't finish properly. If we disable the output panel, there may be other cases besides this one, where the check won't produce any outputs, the output panel won't open, and the user will have no idea why the error highlighting is not working at all.

**Comment by trixnz** [comment]
@Draivin can you show the output of what happens? I've never seen that error before.

**Comment by W4RH4WK** [comment]
@Draivin In my humble opinion when `cargo check` fails because it has not been state which binary (or library) to build, the resulting error should not be put in the output window, but a message in the top of vscode should appear informing the user that something about his current setup is not right.

**Comment by Draivin** [comment]
@trixnz 
![image](https://cloud.githubusercontent.com/assets/947593/18448571/10a20aee-7902-11e6-8b8d-e423af65a1be.png)

**Comment by Draivin** [comment]
@W4RH4WK I agree with you, and this is the general direction we go when dealing with Rusty Code messages. I brought this case up in case someone decided to make a pull request removing the output window without regards to a possible failure.

**Comment by trixnz** [comment]
Ah right, i'm not sure how we would add support for that. Obviously putting it into the workspace preferences would be kinda janky. Maybe we can add a command to set the active `bin`? That would require being able to list off the possible targets though; not sure if cargo supports that.

**Comment by trixnz** [comment]
Actually, it looks like we could use `cargo read-manifest` to list off the targets and provide some kind of window for selecting the one you'd like to target. Although, VSCode doesn't seem to provide a clean way of doing that, AFAIK.

**Comment by W4RH4WK** [comment]
@trixnz For now the `lib` option might suffice, it is not ideal but after all this is more a problem with `cargo check` than with an editor / IDE extension. It may be resolved in the future, see rsolomo/cargo-check#4 and its crossreferences.

**Comment by Draivin** [comment]
@trixnz If you are interested in implementing this, it seems `vscode.window.showQuickPick` works alright:
![image](https://cloud.githubusercontent.com/assets/947593/18449362/3af8cf14-7905-11e6-840c-1cac3063be11.png)

If you are not interested, I probably will look into it soon(™).

**Comment by trixnz** [comment]
I had no idea that even existed, shame on me! I probably won't get time until the weekend, so if I don't see a PR by then I might jump on it 😄

**Comment by smt923** [comment]
Somewhat related to this - I think it would be nice, if possible, to have an option to show the error list instead of compiler output (and have both output and error list hidden on successful builds) on errors, which I foolishly didn't realise existed until today:

http://i.imgur.com/AlAIGZx.png

It's obviously far cleaner and nicer to read through than the raw compiler output, I apologise if this is a setting already or something that's not possible to do with an extension but I can't find anything for this (not even for other languages or global editor stuff)

**Comment by flying-sheep** [comment]
* if an error exists: show error panel 
* else:
	* if this is a check that was triggered via save (`rust.checkOnSave`): show nothing
	* else: show the output panel

**Comment by mitskevich** [comment]
@smt923 @flying-sheep I don't think it's possible to show (not toggle) error panel with current version of VS Code extension API: Microsoft/vscode#11399

**Comment by W4RH4WK** [comment]
@mitskevich That's what I ran into, otherwise I would have added the option.
